bitnami/redis/templates/master/service.yaml

```
@@ -26,6 +26,9 @@ spec:
   {{- if and (eq .Values.master.service.type "LoadBalancer") (not (empty .Values.master.service.loadBalancerIP)) }}
   loadBalancerIP: {{ .Values.master.service.loadBalancerIP }}
   {{- end }}
+  {{- if and (eq .Values.master.service.type "LoadBalancer") (not (empty .Values.master.service.loadBalancerClass)) }}
```

Hello @opencmit2, thanks for the contribution! It is a minor style comment, but this can be simplified a bit:

```suggestion
  {{- if and (eq .Values.master.service.type "LoadBalancer") .Values.master.service.loadBalancerClass }}
```

I think we could make the same modification for the other templates.

---

Hi @FraPazGal 
Although both lines of code are conditional statements in the Helm chart template, they handle .Values.master.service.loadBalancerClass differently. The first line of code checks whether .Values.master.service.loadBalancerClass is not empty, while the second line treats .Values.master.service.loadBalancerClass directly as a boolean value. Therefore, in some cases (such as when .Values.master.service.loadBalancerClass is an empty string), the execution results of the two lines of code may differ.

---

Hi @opencmit2, using `{{- if .Values.foo }}` will resolve to `false` if `foo`  is an empty value and not only if `foo=false`. You can see in [Helm's doc](https://helm.sh/docs/chart_template_guide/control_structures/#ifelse) in which instances Helm considers the above conditional `false`, being an empty string one of them.

You can also double-check this using `helm template` for a [chart already using this](https://github.com/bitnami/charts/blob/3fe40540d4358d60cfd0401bfebc5a031d1234f6/bitnami/mongodb/templates/standalone/svc.yaml#L27):

```
### Using an empty string ###
$ helm template mongodb bitnami/mongodb --set service.type=LoadBalancer,service.loadBalancerClass=""
...
# Source: mongodb/templates/standalone/svc.yaml
...
spec:
  type: LoadBalancer
  allocateLoadBalancerNodePorts: true

### Using a non-empty string ###
$ helm template mongodb bitnami/mongodb --set service.type=LoadBalancer,service.loadBalancerClass=Test
...
# Source: mongodb/templates/standalone/svc.yaml
...
spec:
  type: LoadBalancer
  loadBalancerClass: Test
  allocateLoadBalancerNodePorts: true
```